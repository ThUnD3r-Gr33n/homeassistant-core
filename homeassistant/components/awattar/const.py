"""Constants for the aWATTar integration."""

from __future__ import annotations

import logging
from typing import Final

DOMAIN: Final = "awattar"
LOGGER = logging.getLogger(__package__)
