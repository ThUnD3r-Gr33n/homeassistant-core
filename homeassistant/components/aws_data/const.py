"""Constants for the AWS Data integration."""

import logging

DOMAIN = "aws_data"
DOMAIN_DATA = "aws_client"
CONST_FILTER = "filter"
CONST_SERVICE_NAME = "name"
CONST_ACCOUNT = "account"
CONST_SERVICE_ID = "id"
CONST_SERVICE_REASON = "reason"
CONST_INTERVAL = "interval"
CONST_SECONDS = "seconds"
CONST_AWS_KEY = "aws_key"
CONST_AWS_SECRET = "aws_secret"
CONST_AWS_REGION = "aws_region"
CONST_AWS_SERVICES = "aws_services"
CONST_CE_SELECT = "aws_ce_select"
CONST_SCAN_REGIONS = "aws_scan_regions"
CONST_GENERIC_ID = "aws_id_generic"
CONST_ACCOUNT_ID = "aws_account_id"
CONST_ERRORS = "aws_errors"
SERVICE_EC2 = "ec2"
SERVICE_S3 = "s3"
SERVICE_CE = "ce"
EC2_DEF_INTERVAL = 300
S3_DEF_INTERVAL = 86400
CE_DEF_INTERVAL = 86400

GENERAL_REGION = "us-east-1"
DEFAULT_REGION = GENERAL_REGION

EC2_METRIC_STATISTIC = [
    {"metric": "CPUUtilization", "unit": "Percent"},
    {"metric": "NetworkOut", "unit": "Bytes"},
    {"metric": "EBSWriteBytes", "unit": "Bytes"},
    {"metric": "EBSReadBytes", "unit": "Bytes"},
]
S3_METRIC_STATISTIC = [
    {"metric": "s3_objects", "unit": "count"},
    {"metric": "s3_size", "unit": "Bytes"},
]
CE_METRIC_STATISTIC = [
    {"metric": "monthly_cost", "unit": "sum"},
]
SUPPORTED_SERVICES = [SERVICE_EC2, SERVICE_S3, SERVICE_CE]
SUPPORTED_METRICS = {
    SERVICE_EC2: EC2_METRIC_STATISTIC,
    SERVICE_S3: S3_METRIC_STATISTIC,
    SERVICE_CE: CE_METRIC_STATISTIC,
}

API_DATA = "api"
USER_INPUT = "USER_DATA"
U_ID = "INPUT_ID"
U_SECRET = "INPUT_SECRET"
U_REGIONS = "INPUT_REGIONS"
U_SERVICES = "INPUTE_SERVICES"


_LOGGER = logging.getLogger(__name__)
