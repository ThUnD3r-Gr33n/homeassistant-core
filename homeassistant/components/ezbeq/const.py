"""Constants for the ezbeq Profile Loader integration."""

DOMAIN = "ezbeq"
DEFAULT_PORT = 8080
DEFAULT_NAME = "EzBEQ"
CONF_SOURCE_TYPE = "source_type"
CONF_SOURCE_MEDIA_PLAYER = "source_media_player"
CONF_TMDB_SENSOR = "tmdb_sensor"
CONF_YEAR_SENSOR = "year_sensor"
CONF_CODEC_SENSOR = "codec_sensor"
CONF_CODEC_EXTENDED_SENSOR = "codec_extended_sensor"
CONF_EDITION_SENSOR = "edition_sensor"
CONF_TITLE_SENSOR = "title_sensor"
CONF_PREFERRED_AUTHOR = "preferred_author"
CONF_JELLYFIN_CODEC_SENSOR = "jellyfin_codec_sensor"
CONF_JELLYFIN_DISPLAY_TITLE_SENSOR = "jellyfin_display_title_sensor"
CONF_JELLYFIN_PROFILE_SENSOR = "jellyfin_profile_sensor"
CONF_JELLYFIN_LAYOUT_SENSOR = "jellyfin_layout_sensor"
