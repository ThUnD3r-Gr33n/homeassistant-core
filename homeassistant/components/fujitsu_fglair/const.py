"""Constants for the Fujitsu HVAC (based on Ayla IOT) integration."""

from datetime import timedelta

API_TIMEOUT = 10
API_REFRESH = timedelta(minutes=5)

DOMAIN = "fujitsu_fglair"

CONF_EUROPE = "is_europe"
