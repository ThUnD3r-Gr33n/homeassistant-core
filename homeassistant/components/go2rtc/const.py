"""Go2rtc constants."""

DOMAIN = "go2rtc"
