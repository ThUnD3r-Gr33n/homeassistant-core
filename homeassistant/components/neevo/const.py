"""Constants for the Nee-Vo Tank Monitoring integration."""

from typing import Final

DOMAIN: Final = "neevo"
TANKS: Final = "tanks"
COORDINATOR: Final = "coordinator"
