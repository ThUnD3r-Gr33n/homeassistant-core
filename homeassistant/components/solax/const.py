"""Constants for the solax integration."""

DOMAIN = "solax"

MANUFACTURER = "SolaX Power"

CONF_SOLAX_INVERTER = "solax.inverters"

SOLAX_ENTRY_POINT_GROUP = "solax.inverter"
