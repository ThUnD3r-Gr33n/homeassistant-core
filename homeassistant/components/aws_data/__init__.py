"""The AWS Data integration."""

from __future__ import annotations

from datetime import timedelta
from typing import Any

import voluptuous as vol

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import config_validation as cv
from homeassistant.helpers.typing import ConfigType

from .const import (
    CE_DEF_INTERVAL,
    CONST_ACCOUNT,
    CONST_CE_SELECT,
    CONST_FILTER,
    CONST_INTERVAL,
    CONST_SECONDS,
    CONST_SERVICE_ID,
    CONST_SERVICE_NAME,
    CONST_SERVICE_REASON,
    DOMAIN,
    DOMAIN_DATA,
    EC2_DEF_INTERVAL,
    S3_DEF_INTERVAL,
    SERVICE_CE,
    SERVICE_EC2,
    SERVICE_S3,
    U_REGIONS,
    U_SERVICES,
    USER_INPUT,
)
from .coordinator import (
    AwsDataCEServicesCoordinator,
    AwsDataEC2ServicesCoordinator,
    AwsDataS3ServicesCoordinator,
)

# PLATFORMS: list[Platform] = [Platform.LIGHT]
PLATFORMS: list[Platform] = [Platform.SENSOR]


SERVICE_SCHEMA = vol.Schema(
    {
        vol.Required(CONST_SERVICE_NAME): cv.string,
        vol.Required(CONST_ACCOUNT): cv.string,
        vol.Required(CONST_SERVICE_ID): vol.All(cv.ensure_list, [cv.string]),
        vol.Optional(CONST_SERVICE_REASON, default="Exclude"): cv.string,
    }
)

CONFIG_SCHEMA = vol.Schema(
    {
        DOMAIN: vol.Schema(
            {
                vol.Optional(CONST_FILTER): vol.All(cv.ensure_list, [SERVICE_SCHEMA]),
                vol.Optional(CONST_INTERVAL): vol.All(
                    cv.ensure_list,
                    [
                        vol.Schema(
                            {
                                vol.Required(CONST_SERVICE_NAME): cv.string,
                                vol.Required(CONST_SECONDS): int,
                            }
                        )
                    ],
                ),
            }
        )
    },
    extra=vol.ALLOW_EXTRA,
)


async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
    """Set up AWS Data Filter."""
    conf = config[DOMAIN]
    if conf == hass.data.get(DOMAIN_DATA, None):
        return True

    hass.data[DOMAIN_DATA] = conf
    entries = hass.config_entries.async_entries(DOMAIN)
    for entr in entries:
        hass.config_entries.async_update_entry(entry=entr, data=entr.data)

    return True


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Set up AWS Data from a config entry."""

    user_data = dict(entry.data)
    selected_regions = user_data[USER_INPUT][U_REGIONS]
    services = user_data[USER_INPUT][U_SERVICES]
    store_coord: dict[str, Any] = {}

    intervals = {}
    domain_data = hass.data.get(DOMAIN_DATA, {})
    for data in domain_data.get(CONST_INTERVAL, []):
        intervals[data[CONST_SERVICE_NAME]] = {CONST_SECONDS: data[CONST_SECONDS]}

    if SERVICE_EC2 in services:
        ec2_interval = intervals.get(SERVICE_EC2, {})
        store_coord[SERVICE_EC2] = ec2_coordinator = AwsDataEC2ServicesCoordinator(
            hass,
            entry=entry,
            regions=selected_regions,
            services=services,
            update_interval=timedelta(
                seconds=ec2_interval.get(CONST_SECONDS, EC2_DEF_INTERVAL)
            ),
        )
        await ec2_coordinator.async_config_entry_first_refresh()

    if SERVICE_S3 in services:
        s3_interval = intervals.get(SERVICE_S3, {})
        store_coord[SERVICE_S3] = s3_coordinator = AwsDataS3ServicesCoordinator(
            hass,
            entry=entry,
            regions=selected_regions,
            services=services,
            update_interval=timedelta(
                seconds=s3_interval.get(CONST_SECONDS, S3_DEF_INTERVAL)
            ),
        )
        await s3_coordinator.async_config_entry_first_refresh()

    if user_data[USER_INPUT][CONST_CE_SELECT]:
        ce_interval = intervals.get(SERVICE_S3, {})
        store_coord[SERVICE_CE] = ce_coordinator = AwsDataCEServicesCoordinator(
            hass,
            entry=entry,
            regions=selected_regions,
            services=services,
            update_interval=timedelta(
                seconds=ce_interval.get(CONST_SECONDS, CE_DEF_INTERVAL)
            ),
        )
        await ce_coordinator.async_config_entry_first_refresh()

    entry.runtime_data = {"coord": store_coord}
    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload a config entry."""

    entry.runtime_data = {}
    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
