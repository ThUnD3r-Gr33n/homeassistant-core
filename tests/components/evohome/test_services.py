"""The tests for evohome services.

Assert service calls my mocking the HTTP GET/PUTs and checking the URL/payloads.

The payload (json) of some requests (to the vendor API) are difficult to assert
because they contain datetimes relative to a particular zone's scheduled setpoint.
Instead, they are collected here, so that they can be simply asserted via snapshot.
"""

from __future__ import annotations

from unittest.mock import patch

from freezegun.api import FrozenDateTimeFactory
import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.climate import (
    ATTR_HVAC_MODE,
    ATTR_TEMPERATURE,
    SERVICE_SET_HVAC_MODE,
    SERVICE_SET_TEMPERATURE,
    HVACMode,
)
from homeassistant.components.evohome import DOMAIN, EvoService
from homeassistant.const import ATTR_ENTITY_ID, Platform
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import HomeAssistantError, ServiceValidationError

from .conftest import entity_of_ctl, entity_of_zone, setup_evohome
from .const import TEST_INSTALLS


@pytest.mark.parametrize("install", TEST_INSTALLS)
async def test_evohome_svcs_ctl(
    hass: HomeAssistant,
    config: dict[str, str],
    freezer: FrozenDateTimeFactory,
    snapshot: SnapshotAssertion,
    install: str,
) -> None:
    """Test domain-specific services of a evohome-compatible controller."""

    freezer.move_to("2024-07-10T12:00:00Z")
    results = []

    async for _ in setup_evohome(hass, config, install=install):
        services = list(hass.services.async_services_for_domain(DOMAIN))
        assert services == snapshot

        with patch("evohomeasync2.location.Location.refresh_status") as mock_fcn:
            await hass.services.async_call(
                DOMAIN,
                EvoService.REFRESH_SYSTEM,
                service_data=None,
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == ()
            assert mock_fcn.await_args.kwargs == {}

        system_mode = "Off" if install == "h032585" else "HeatingOff"

        with patch("evohomeasync2.controlsystem.ControlSystem.set_mode") as mock_fcn:
            await hass.services.async_call(
                DOMAIN,
                EvoService.SET_SYSTEM_MODE,
                service_data={"mode": system_mode},
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == (system_mode,)
            assert mock_fcn.await_args.kwargs == {"until": None}

        if install == "h032585":  # only supports Heat, Off
            return results

        with patch("evohomeasync2.controlsystem.ControlSystem.set_mode") as mock_fcn:
            await hass.services.async_call(
                DOMAIN,
                EvoService.SET_SYSTEM_MODE,
                service_data={"mode": "Away", "period": {"days": 21}},
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == ("Away",)
            assert "until" in mock_fcn.await_args.kwargs

            results.append(mock_fcn.await_args.kwargs)

        with patch("evohomeasync2.controlsystem.ControlSystem.set_mode") as mock_fcn:
            await hass.services.async_call(
                DOMAIN,
                EvoService.SET_SYSTEM_MODE,
                service_data={"mode": "AutoWithEco", "duration": {"minutes": 150}},
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == ("AutoWithEco",)
            assert "until" in mock_fcn.await_args.kwargs

            results.append(mock_fcn.await_args.kwargs)

        if EvoService.RESET_SYSTEM not in services:
            return results

        with patch("evohomeasync2.controlsystem.ControlSystem.set_mode") as mock_fcn:
            # mock_refresh.side_effect =

            await hass.services.async_call(
                DOMAIN,
                EvoService.RESET_SYSTEM,
                service_data=None,
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == ("AutoWithReset",)
            assert mock_fcn.await_args.kwargs == {"until": None}

    assert results == snapshot  # noqa: RET503


@pytest.mark.parametrize("install", TEST_INSTALLS)
async def test_evohome_svcs_zon(
    hass: HomeAssistant,
    config: dict[str, str],
    freezer: FrozenDateTimeFactory,
    snapshot: SnapshotAssertion,
    install: str,
) -> None:
    """Test domain-specific services of a evohome-compatible climate zone."""

    freezer.move_to("2024-07-10T12:00:00Z")
    results = []

    async for _ in setup_evohome(hass, config, install=install):
        zone = entity_of_zone(hass)  # the first zone will do...

        # reset zone to follow_schedule
        with patch("evohomeasync2.zone.Zone.reset_mode") as mock_fcn:
            await hass.services.async_call(
                DOMAIN,
                EvoService.RESET_ZONE_OVERRIDE,
                service_data={"entity_id": zone.entity_id},
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == ()
            assert mock_fcn.await_args.kwargs == {}

        # set zone to temporary_override
        with patch("evohomeasync2.zone.Zone.set_temperature") as mock_fcn:
            await hass.services.async_call(
                DOMAIN,
                EvoService.SET_ZONE_OVERRIDE,
                service_data={
                    "entity_id": zone.entity_id,
                    "setpoint": 21.3,
                    "duration": {"minutes": 150},
                },
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == (21.3,)
            assert "until" in mock_fcn.await_args.kwargs

            results.append(mock_fcn.await_args.kwargs)

        # set zone to temporary_override (advance to next scheduled setpoint)
        with patch("evohomeasync2.zone.Zone.set_temperature") as mock_fcn:
            await hass.services.async_call(
                DOMAIN,
                EvoService.SET_ZONE_OVERRIDE,
                service_data={
                    "entity_id": zone.entity_id,
                    "setpoint": 19.7,
                },
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == (19.7,)
            assert (
                "until" in mock_fcn.await_args.kwargs
            )  # until next scheduled setpoint

            results.append(mock_fcn.await_args.kwargs)  # varies by install fixture/zone

        # # set zone to permanent_override
        # with patch("evohomeasync2.zone.Zone.set_temperature") as mock_fcn:
        # await hass.services.async_call(
        #     DOMAIN,
        #     EvoService.SET_ZONE_OVERRIDE,
        #     service_data={
        #         "entity_id": zone.entity_id,
        #         "setpoint": 19.7,
        #         "duration": None,
        #     },
        #     blocking=True,
        # )

        # assert mock_fcn.await_count == 1
        # assert mock_fcn.await_args.args == (19.7,)
        # assert "until" in mock_fcn.await_args.kwargs  # until next scheduled setpoint

        # results.append(mock_fcn.await_args.kwargs)  # varies by install fixture/zone

    assert results == snapshot


@pytest.mark.parametrize("install", TEST_INSTALLS)
async def test_climate_svcs_ctl(
    hass: HomeAssistant,
    config: dict[str, str],
    freezer: FrozenDateTimeFactory,
    snapshot: SnapshotAssertion,
    install: str,
) -> None:
    """Test climate services of a evohome-compatible controller."""

    freezer.move_to("2024-07-10T12:00:00Z")
    results = []

    async for _ in setup_evohome(hass, config, install=install):
        ctl = entity_of_ctl(hass)

        with patch("evohomeasync2.controlsystem.ControlSystem.set_mode") as mock_fcn:
            await hass.services.async_call(
                Platform.CLIMATE,
                SERVICE_SET_HVAC_MODE,
                {
                    ATTR_ENTITY_ID: ctl.entity_id,
                    ATTR_HVAC_MODE: HVACMode.OFF,
                },
                blocking=True,
            )

            system_mode = "Off" if install == "h032585" else "HeatingOff"

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == (system_mode,)
            assert mock_fcn.await_args.kwargs == {"until": None}

        with patch("evohomeasync2.controlsystem.ControlSystem.set_mode") as mock_fcn:
            await hass.services.async_call(
                Platform.CLIMATE,
                SERVICE_SET_HVAC_MODE,
                {
                    ATTR_ENTITY_ID: ctl.entity_id,
                    ATTR_HVAC_MODE: HVACMode.HEAT,
                },
                blocking=True,
            )

            system_mode = "Heat" if install == "h032585" else "Auto"

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == (system_mode,)
            assert mock_fcn.await_args.kwargs == {"until": None}

        for hvac_mode in HVACMode:
            if hvac_mode in (HVACMode.HEAT, HVACMode.OFF):
                continue

            with pytest.raises(HomeAssistantError):
                await hass.services.async_call(
                    Platform.CLIMATE,
                    SERVICE_SET_HVAC_MODE,
                    {
                        ATTR_ENTITY_ID: ctl.entity_id,
                        ATTR_HVAC_MODE: hvac_mode,
                    },
                    blocking=True,
                )

    assert results == snapshot


@pytest.mark.parametrize("install", TEST_INSTALLS)
async def test_water_heater_svcs_dhw(
    hass: HomeAssistant,
    config: dict[str, str],
    freezer: FrozenDateTimeFactory,
    snapshot: SnapshotAssertion,
    install: str,
) -> None:
    """Test water_heater services of a evohome-compatible DHW zone."""

    freezer.move_to("2024-07-10T12:00:00Z")
    results = []

    async for _ in setup_evohome(hass, config, install=install):
        zone = entity_of_zone(hass)  # the first zone will do...

        with patch("evohomeasync2.zone.Zone.set_temperature") as mock_fcn:
            await hass.services.async_call(
                Platform.CLIMATE,
                SERVICE_SET_TEMPERATURE,
                {
                    ATTR_ENTITY_ID: zone.entity_id,
                    ATTR_TEMPERATURE: 21.5,
                },
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == (21.5,)
            assert (
                "until" in mock_fcn.await_args.kwargs
            )  # until next scheduled setpoint

            results.append(mock_fcn.await_args.kwargs)  # varies by install fixture/zone

        for temperature in (0, 40):
            with pytest.raises(ServiceValidationError):
                await hass.services.async_call(
                    Platform.CLIMATE,
                    SERVICE_SET_TEMPERATURE,
                    {
                        ATTR_ENTITY_ID: zone.entity_id,
                        ATTR_TEMPERATURE: temperature,
                    },
                    blocking=True,
                )

    assert results == snapshot


@pytest.mark.parametrize("install", TEST_INSTALLS)
async def test_climate_svcs_zon(
    hass: HomeAssistant,
    config: dict[str, str],
    freezer: FrozenDateTimeFactory,
    snapshot: SnapshotAssertion,
    install: str,
) -> None:
    """Test climate services of a evohome-compatible climate zone."""

    freezer.move_to("2024-07-10T12:00:00Z")
    results = []

    async for _ in setup_evohome(hass, config, install=install):
        zone = entity_of_zone(hass)  # the first zone will do...

        with patch("evohomeasync2.zone.Zone.set_temperature") as mock_fcn:
            await hass.services.async_call(
                Platform.CLIMATE,
                SERVICE_SET_TEMPERATURE,
                {
                    ATTR_ENTITY_ID: zone.entity_id,
                    ATTR_TEMPERATURE: 21.5,
                },
                blocking=True,
            )

            assert mock_fcn.await_count == 1
            assert mock_fcn.await_args.args == (21.5,)
            assert (
                "until" in mock_fcn.await_args.kwargs
            )  # until next scheduled setpoint

            results.append(mock_fcn.await_args.kwargs)  # varies by install fixture/zone

        for temperature in (0, 40):
            with pytest.raises(ServiceValidationError):
                await hass.services.async_call(
                    Platform.CLIMATE,
                    SERVICE_SET_TEMPERATURE,
                    {
                        ATTR_ENTITY_ID: zone.entity_id,
                        ATTR_TEMPERATURE: temperature,
                    },
                    blocking=True,
                )

    assert results == snapshot
