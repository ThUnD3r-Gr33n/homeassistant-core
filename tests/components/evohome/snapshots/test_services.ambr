# serializer version: 1
# name: test_climate_svcs_ctl[default]
  list([
  ])
# ---
# name: test_climate_svcs_ctl[h032585]
  list([
  ])
# ---
# name: test_climate_svcs_ctl[h099625]
  list([
  ])
# ---
# name: test_climate_svcs_ctl[minimal]
  list([
  ])
# ---
# name: test_climate_svcs_ctl[sys_004]
  list([
  ])
# ---
# name: test_climate_svcs_zon[default]
  list([
    dict({
      'until': datetime.datetime(2024, 7, 10, 21, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_climate_svcs_zon[h032585]
  list([
    dict({
      'until': datetime.datetime(2024, 7, 10, 21, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_climate_svcs_zon[h099625]
  list([
    dict({
      'until': datetime.datetime(2024, 7, 10, 19, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_climate_svcs_zon[minimal]
  list([
    dict({
      'until': datetime.datetime(2024, 7, 10, 21, 10, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_climate_svcs_zon[sys_004]
  list([
    dict({
      'until': None,
    }),
  ])
# ---
# name: test_evohome_svcs_ctl[default]
  list([
    'refresh_system',
    'reset_system',
    'set_system_mode',
    'clear_zone_override',
    'set_zone_override',
  ])
# ---
# name: test_evohome_svcs_ctl[default].1
  list([
    dict({
      'until': HAFakeDatetime(2024, 7, 30, 23, 0, tzinfo=datetime.timezone.utc),
    }),
    dict({
      'until': HAFakeDatetime(2024, 7, 10, 14, 30, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_evohome_svcs_ctl[h032585]
  list([
    'refresh_system',
    'set_system_mode',
    'clear_zone_override',
    'set_zone_override',
  ])
# ---
# name: test_evohome_svcs_ctl[h099625]
  list([
    'refresh_system',
    'set_system_mode',
    'clear_zone_override',
    'set_zone_override',
  ])
# ---
# name: test_evohome_svcs_ctl[minimal]
  list([
    'refresh_system',
    'reset_system',
    'set_system_mode',
    'clear_zone_override',
    'set_zone_override',
  ])
# ---
# name: test_evohome_svcs_ctl[minimal].1
  list([
    dict({
      'until': HAFakeDatetime(2024, 7, 30, 23, 0, tzinfo=datetime.timezone.utc),
    }),
    dict({
      'until': HAFakeDatetime(2024, 7, 10, 14, 30, tzinfo=datetime.timezone.utc),
    }),
  ])
# ---
# name: test_evohome_svcs_ctl[sys_004]
  list([
    'refresh_system',
    'set_system_mode',
    'clear_zone_override',
    'set_zone_override',
  ])
# ---
# name: test_evohome_svcs_zon[default]
  list([
    dict({
      'until': HAFakeDatetime(2024, 7, 10, 14, 30, tzinfo=datetime.timezone.utc),
    }),
    dict({
      'until': None,
    }),
  ])
# ---
# name: test_evohome_svcs_zon[h032585]
  list([
    dict({
      'until': HAFakeDatetime(2024, 7, 10, 14, 30, tzinfo=datetime.timezone.utc),
    }),
    dict({
      'until': None,
    }),
  ])
# ---
# name: test_evohome_svcs_zon[h099625]
  list([
    dict({
      'until': HAFakeDatetime(2024, 7, 10, 14, 30, tzinfo=datetime.timezone.utc),
    }),
    dict({
      'until': None,
    }),
  ])
# ---
# name: test_evohome_svcs_zon[minimal]
  list([
    dict({
      'until': HAFakeDatetime(2024, 7, 10, 14, 30, tzinfo=datetime.timezone.utc),
    }),
    dict({
      'until': None,
    }),
  ])
# ---
# name: test_evohome_svcs_zon[sys_004]
  list([
    dict({
      'until': HAFakeDatetime(2024, 7, 10, 14, 30, tzinfo=datetime.timezone.utc),
    }),
    dict({
      'until': None,
    }),
  ])
# ---
